amm/drink-tests/src/router_tests.rs

```diff
@@ -283,7 +284,173 @@ fn test_fees(mut session: Session) {
+    psp22_utils::transfer(&mut session, ice.into(), ice_wood_pair.into(), amount, BOB).unwrap();
+    psp22_utils::transfer(&mut session, wood.into(), ice_wood_pair.into(), amount, BOB).unwrap();
+
+    let liquidity_minted = session
+        .execute(ice_wood_pair.mint(bob()))
+        .unwrap()
+        .result
+        .unwrap()
+        .unwrap();
+
+    let bob_lp_balance: u128 = psp22_utils::balance_of(&mut session, ice_wood_pair.into(), bob());
+    assert_eq!(liquidity_minted, bob_lp_balance);
+
+    let deadline = now + 10;
+
+    // Perform swap
+    let swap_amount = 10_000 * TOKEN;
+    psp22_utils::increase_allowance(&mut session, ice.into(), router.into(), swap_amount, BOB)
+        .unwrap();
+    let swap_res = session
+        .execute(router.swap_exact_tokens_for_tokens(
+            swap_amount,
+            0,
+            vec![ice.into(), wood.into()],
+            bob(),
+            deadline,
+        ))
+        .unwrap()
+        .result
+        .unwrap()
+        .unwrap();
+
+    // We cannot assert that as it depends on the pool's liquidity and swap amount. The price will move from the spot price.
+    // assert!(swap_res[1] >= min_amount_out);
+
+    // No fees distributed until the burn/mint transaction.
+    assert!(psp22_utils::balance_of(&mut session, ice_wood_pair.into(), charlie()) == 0);
+    // Burn some liquidity to trigger fee collection.
+    psp22_utils::increase_allowance(
+        &mut session,
+        ice_wood_pair.into(),
+        router.into(),
+        liquidity_minted,
+        BOB,
+    )
+    .unwrap();
+    let (_ice, _wood) = session
+        .execute(router.remove_liquidity(
+            ice.into(),
+            wood.into(),
+            liquidity_minted,
+            0,
+            0,
+            bob(),
+            deadline,
+        ))
+        .unwrap()
+        .result
+        .unwrap()
+        .unwrap();
+
+    // Fees now sent to `fee_to` address (CHARLIE).
+    let charlie_lp_balance = psp22_utils::balance_of(&mut session, ice_wood_pair.into(), charlie());
+    // Charlie withdraws his fees
+    psp22_utils::increase_allowance(
+        &mut session,
+        ice_wood_pair.into(),
+        router.into(),
+        charlie_lp_balance,
+        CHARLIE,
+    )
+    .unwrap();
+    let (protocol_fees_ice, protocol_fees_wood) = router::remove_liquidity(
+        &mut session,
+        router.into(),
+        ice.into(),
+        wood.into(),
+        charlie_lp_balance,
+        0,
+        0,
+        charlie(),
+    );
+    // We cannot assert exactly how much fees Charlie will get, due to roundings etc,
+    // but it should be close to the expected value
+    let received_protocol_fees = protocol_fees_ice + protocol_fees_wood;
+
+    // Trading fee is 0.3%.
+    let trading_fee = (session
+        .query(ice_wood_pair.get_fee())
+        .unwrap()
+        .result
+        .unwrap() as f64)
+        / 1000.0;
 
     // Protocol fees are 1/6 of the trading fees.
```

Comment:
The issue is that 1/6 was the correct value for 0.3% trading fee (0.05% protocol fee from 0.3% trading fee). Now that it's custom - 0.1% - the actual "protocol cut" is 1/2 (0.05% from 0.1%).

From the whitepaper:
> If the `feeTo` address is set, the protocol will begin charging a 5-basis-point fee, which is taken as a 1/6 cut of the 30-basis-point fees earned by liquidity providers.

Another point - originally, this test was written to test the protocol fee collection, ie whether we mint the correct amount of tokens for the protocol.

Reply:
This is a little fishy, my understanding is that it takes 1/6 of the collected fees, no matter what they are. So 0.05% is not universally true...